governance/program/src/instruction.rs

```diff
@@ -0,0 +1,236 @@
+//! Program instructions
+
+use solana_program::{epoch_schedule::Slot, instruction::Instruction};
+
+use crate::state::Vote;
+
+/// Instructions supported by the Governance program
+#[derive(Clone)]
+#[allow(clippy::large_enum_variant)]
+pub enum GovernanceInstruction {
+    /// Creates Governance Realm account which aggregates governances for given Governance Mint and optional Council Mint.
+    ///
+    /// 1. `[writable]` Governance Realm account.
+    /// 2. `[]` Governance Token Mint.
+    /// 3. `[writable, signer]` Governances Token Holding account.
+    /// 4. `[signer]` Payer.
+    /// 5. `[]` System.
+    /// 6. `[]` SPL Token.
+    /// 7. `[]` Sysvar Rent.
+    /// 8. `[]` Council Token mint - optional.
+    /// 9. `[writable, signer]` Council Token Holding account - optional.
+    CreateGovernanceRealm {
+        /// UTF-8 encoded Governance Realm name.
+        name: String,
+    },
+
+    /// Deposits governing tokens (Governance or Council) to Governance Realm and establishes your voter weight to be used for voting within the Realm.
+    /// Note: If subsequent (top up) deposit is made and there are active votes for the Voter then the weights won't be updated automatically.
+    /// It can be done by relinquishing votes on active Proposals and voting again with the new weight.
+    ///
+    ///  0. `[]` Governance Realm account.
+    ///  1. `[]` Governing Token mint.
+    ///  2. `[writable]` Governing Token Holding account.
+    ///  3. `[writable]` Governing Token Source account.
+    ///  4. `[writable]` Voter Record account.
+    ///  4. `[signer]` Voter (payer).
+    ///  5. `[]` System.
+    ///  6. `[]` SPL Token.
+    ///  7. `[]` Sysvar Rent.
+    DepositGoverningTokens {
+        /// Amount of Governing tokens to deposit. If None then all tokens from Source Account would be deposited.
+        amount: Option<u64>,
+    },
+
+    /// Withdraws governing tokens (Governance or Council) from Governance Realm and updates your voter weight within the Realm.
+    /// Note: It's only possible to withdraw tokens if the Voter doesn't have any outstanding active votes.
+    /// If there are any outstanding votes then they must be relinquished before tokens could be withdrawn.
+    ///
+    ///  0. `[]` Governance Realm account.
+    ///  1. `[]` Governing Token mint.
+    ///  2. `[writable]` Governing Token Holding account.
+    ///  3. `[writable]` Governing Token Source account.
+    ///  4. `[writable]` Voter Record account.
+    ///  6. `[]` SPL Token.
+    ///  7. `[]` Sysvar Rent.   
+    WithdrawGoverningTokens {
+        /// Amount of Governing tokens to withdraw. If None then all Voter tokens from Holding Account would be withdrawn.
+        amount: Option<u64>,
```

Comment:
I'm not sure if there is a case for that. In theory you could want to transfer some of your governing tokens but not all to somebody else.
Since I'm not sure it's needed I removed it for now